backend/catalogue/models.py

```diff
@@ -114,7 +144,417 @@ def server_info(self):
+                'format': 'image/png',
+                'transparent': True,
+            }
+            return requests.get(url=self.server_url, params=params).url
+
+    def geoserver_legend_rule_to_legend_key(self, legend_rule: dict) -> dict:
+        """
+        Converts a GeoServer legend rule into a legend key dictionary.
+
+        Args:
+            legend_rule (dict): A dictionary from a GeoServer legend's rules.
+
+        Returns:
+            dict: A dictionary representing a legend key with a `label` and `style`.
+
+        Raises:
+            ValueError: If the legend rule's symbolizer type is unsupported.
+
+        Example:
+            >>> self.geoserver_legend_rule_to_legend_key(geoserver_legend['Legend'][0]['rules'][0])
+            {
+                'label': 'Red Polygon',
+                'style': {
+                    'height': '100%',
+                    'width': '100%',
+                    'backgroundColor': '#ff0000',
+                    'border': '2px solid #000000'
+                }
+            }
+
+        Note:
+        - The method handles 'Polygon' and 'Point' symbolizer types. Symbolizer types not
+            supported by this method will raise a `ValueError`.
+        """
+        symbolizer_type = next(iter(legend_rule['symbolizers'][0]))
+        symbolizer = legend_rule['symbolizers'][0][symbolizer_type]
+        style = {}
+
+        # Handle Polygon symbolizer type
+        if symbolizer_type == 'Polygon':
+            # Construct a legend key dictionary
+            style = {
+                'height': '100%',
+                'width': '100%',
+            }
+            
+            # Set background color to 'fill' if present
+            if 'fill' in symbolizer:
+                style['backgroundColor'] = symbolizer['fill']
+
+            # Set border to stroke width and color if present
+            if ('stroke-width' in symbolizer) or ('stroke' in symbolizer):
+                style['border'] = f"{symbolizer.get('stroke-width', 2)}px solid {symbolizer.get('stroke', 'black')}"
+
+        # Handle Point symbolizer type
+        elif symbolizer_type == 'Point':
+            # Construct a legend key dictionary
+            style = {
+                'height': f"{symbolizer['size']}px",
+                'width': f"{symbolizer['size']}px",
+            }
+
+            graphic = symbolizer['graphics'][0]
+
+            # Set background color to 'fill' if present
+            if 'fill' in graphic:
+                style['backgroundColor'] = graphic['fill']
+
+            # Set border to stroke width and color if present
+            if ('stroke-width' in graphic) or ('stroke' in graphic):
+                style['border'] = f"{graphic.get('stroke-width', 2)}px solid {graphic.get('stroke', 'black')}"
+
+            # Custom style for circle markers
+            if 'mark' in graphic and graphic['mark'] == 'circle':
+                style['borderRadius'] = '100%'
```

Comment:
This a general thought, in the context of that "Philosophy of Software Design" book I mentioned last team meeting.  (and I do like the docstrings and the generally-improved documentation!).  One of things mentioned, obvious once you've heard it, is "comments that just repeat the code aren't useful".  So, this is arguably a case of that.  What might be more useful is a bit of context that isn't immediately obvious from the following lines; eg instead of "set border..." it might be a note that a border can be defined in one of two ways so we need to handle both... that sort of thing.

(I think it's super hard to teach commenting to first-years, because they don't have the context of what is unexpected, and you just get a bunch "i++; // add one to i")

Reply:
I tried some improvement (here at least) with b7ab443